### PR TITLE
[ROX-11280] : Add decommissioned cluster retention config and default value

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

- Added `decommissioned_cluster_retention_duration_days` config to config proto
- Added 90 days default value for retention config

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

- No upgrade steps needed for this change
- Autogenerated API Reference doc shows new retention config under `ConfigService`
- Change log and additional documentation will be added after adding the cluster auto removal mechanism

## Testing Performed

Manually called the ConfigService API to see if `decommissionedClusterRetentionDurationDays` field exists and is set to expected default value

D50F03EC-742B-4BA8-819C-E5E56D3FB1C0_4_5005_c

### CODE REVIEW

File: proto/storage/config.proto [hunk 1]
```diff
@@ -43,15 +43,16 @@ message AlertRetentionConfig {
...
-        int32                DEPRECATED_alert_retention_duration_days      = 1;
-        AlertRetentionConfig alert_config                                  = 3;
+        int32                DEPRECATED_alert_retention_duration_days       = 1;
+        AlertRetentionConfig alert_config                                   = 3;
     }
-    int32                    image_retention_duration_days                 = 2;
-    int32                    expired_vuln_req_retention_duration_days      = 4;
+    int32                    image_retention_duration_days                  = 2;
+    int32                    expired_vuln_req_retention_duration_days       = 4;
+    int32                    decommissioned_cluster_retention_duration_days = 5;
```
Comment (md2119) [inline proto/storage/config.proto:55]: Let's see if we get something that beats this var name length

**Review by clickboo — APPROVED** [review]

File: proto/storage/config.proto (hunk 1 above)
Comment (pedrottimark) [reply proto/storage/config.proto:55]: What is plus and minus of uncontacted versus decomissioned, from viewpoint of some who reads API Reference without the context that we have from discussion of feature?

Comment (pedrottimark) [reply proto/storage/config.proto:55]: Or does that imply never contacted aka uninitialized?

Comment (clickboo) [reply proto/storage/config.proto:55]: To me that: uncontacted is not as intentional. Could be inferred as uninitialized.

**Pull request merged**